runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -555,7 +556,7 @@ public synchronized void invalidateLayout() {
                 try {
                     Thread.sleep(retryRate * 1000);
                 } catch (InterruptedException e) {
-                    e.printStackTrace();
+                    throw new UnrecoverableCorfuException("Interrupted while fetching layout", e);
```

Comment:
Isn't it best practice to re-interrupt the thread? We don't know what kind of mechanism upstream can swallow this exception or engage in funky thread-related behaviors. When the interrupted exception is caught, the interrupted flag is set to false. It is probably safer to reset it to true by interrupting again.

This applies to all these cases.

Reply:
Hm, okay good point. I created a new error type that will reset the interrupt pending flag as well.